src/components/trade/PriceImpact/index.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react'
+import BigNumber from 'bignumber.js'
+import BN from 'bn.js'
+import { TokenDex } from '@gnosis.pm/dex-js'
+
+import { BoldColourTag } from './PriceImpact.styled'
+import { FormMessage } from 'components/common/FormMessage'
+import { HelpTooltip, HelpTooltipContainer } from 'components/Tooltip'
+import { PriceSuggestionsWrapper } from '../PriceSuggestions/PriceSuggestions'
+
+import useBestAsk from 'hooks/useBestAsk'
+
+import { formatSmart, parseAmount } from 'utils'
+import { calculatePriceImpact, determinePriceWarning, getImpactColourClass } from './utils'
+
+export interface PriceImpactArgsBase {
+  limitPrice?: string | null
+  bestAskPrice: BigNumber | null
+}
+
+export interface PriceImpactArgs extends PriceImpactArgsBase {
+  fillPrice: BigNumber | null
+}
+
+interface PriceImpactProps {
+  baseToken: TokenDex
+  quoteToken: TokenDex
+  limitPrice?: string | null
+  fillPrice: BigNumber | null
+  networkId: number
+}
+
+const PriceImpactTooltip: React.FC = () => (
+  <HelpTooltipContainer>
+    The difference between the market price and the limit price due to order size
+  </HelpTooltipContainer>
+)
+
+function PriceImpact({
+  limitPrice,
+  fillPrice,
+  baseToken,
+  quoteToken,
+  networkId,
+}: PriceImpactProps): React.ReactElement | null {
+  const { id: baseTokenId } = baseToken
+  const { id: quoteTokenId } = quoteToken
+
+  // TODO: useBestAskPrice hook here
```

Comment:
No you do, right? do we need this TODO?

Reply:
is `useBestAskPrice` and `useBestAsk` the same?

Reply:
yeah theyre the same

Reply:
and yeah it's already in, will remove comment